app/controllers/sessions_controller.rb

```diff
@@ -47,6 +47,12 @@ def confirm_email
     if new_user?
       user = User.create(username: session[:username], email: params[:email])
 
+      # If a new user logged in with Google, set email_for_google to the
+      # email address they used to log in.
+      if session[:email_for_google]
```

Comment:
Stylistic nit: in Ruby it is typically preferred to be explicit with this kind of check, so that the reader can tell at a glance if you want `session[:email_for_google]` to be `true` or to be not-nil, or whatever. So I would write this instead as `if session[:email_for_google].present?` which is functionally identical to your code, but reads as a more explicit description of your intent "if we have a session[:email_for_google], then...".

Reply:
Makes sense -- done.

Style feedback very much appreciated! I am mostly pattern-matching on the rest of the codebase and don't know much about Ruby language features or style -- learning along the way.